server/app/views/BaseHtmlView.java

```diff
@@ -68,12 +68,21 @@ protected static ButtonTag submitButton(String id, String textContents) {
   }
 
   protected static ButtonTag redirectButton(String id, String text, String redirectUrl) {
-    return asRedirectButton(
+    return asRedirectElement(
         TagCreator.button(text).withId(id).withClasses(Styles.M_2), redirectUrl);
   }
 
-  protected static ButtonTag asRedirectButton(ButtonTag buttonEl, String redirectUrl) {
-    return buttonEl.attr("onclick", String.format("window.location = '%s';", redirectUrl));
+  /**
+   * Turns provided element into a clickable element. Upon click the user will be redirected to the
+   * provided url. It's up to caller of this method to style element appropriately to make it clear
+   * that the element is clickable. For example add hover effect and change cursor style.
+   *
+   * @return The element itself.
+   */
+  protected static <T extends Tag> T asRedirectElement(T element, String redirectUrl) {
+    // Attribute `data-redirect-to` is handled in JS by main.ts file.
+    element.attr("data-redirect-to", redirectUrl);
```

Comment:
Not blocking: Can save a line by returning the result of element.attr(), since J2HTML elements support chaining.

Reply:
That how it was initially. But with introduction of T it no longer works. Some generics shenanigans because `.attr()` also returns generic T according to function declaration and compiler somehow things that it returns `Tag`